cli/azd/pkg/infra/provisioning/bicep/bicep_provider_test.go

```diff
@@ -493,6 +518,29 @@ func prepareDestroyMocks(mockContext *mocks.MockContext) {
 		}, nil
 	})
 
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.Method == http.MethodGet && strings.Contains(request.URL.Path, "/vaults/kv2-123")
+	}).RespondFn(func(request *http.Request) (*http.Response, error) {
```

Comment:
Something to consider in the future - since there's a lot of duplication here - we could always add something like:

```go
func respondFnForKeyVault(name string) func(request *http.Request) (*http.Response, error) {
    return func(request *http.Request) (*http.Response, error) {
		keyVaultResponse := armkeyvault.VaultsClientGetResponse{
			Vault: armkeyvault.Vault{
				ID:       convert.RefOf(name ),
				Name:     convert.RefOf(name),
				Location: convert.RefOf("eastus2"),
				Properties: &armkeyvault.VaultProperties{
					EnableSoftDelete:      convert.RefOf(true),
					EnablePurgeProtection: convert.RefOf(false),
				},
			},
		}

		keyVaultBytes, _ := json.Marshal(keyVaultResponse)

		return &http.Response{
			StatusCode: http.StatusOK,
			Body:       io.NopCloser(bytes.NewBuffer(keyVaultBytes)),
		}, nil
    }
}
```

And then do something like:

```go
mockContext.HttpClient.When(func(request *http.Request) bool {
		return request.Method == http.MethodGet && strings.Contains(request.URL.Path, "/vaults/kv2-123")
	}).RespondFn(respondFnForKeyVault("kv2-123")
```

As a way to remove the duplication.

Reply:
You may want to do something similar to what I have in https://github.com/Azure/azure-dev/blob/main/cli/azd/test/mocks/graphsdk/mocks.go.

We call these graph mocks in multiple places with multiple values and creating a single method that can handle the variations for testing will allow more cleaner or organized code.